### PR TITLE
introduce new plugin for jumping git root

### PR DESCRIPTION
**Pull request by PatrickF1**
introduce new plugin for jumping to the root of current git repo
tested locally

### CODE REVIEW

File: plugins/README.md [hunk 1]
```diff
@@ -29,7 +29,8 @@ Plugins extend the capabilities of `nnn`. They are _executable_ scripts (or bina
 | [fzopen](fzopen) | Fuzzy find file(s) in subtree to edit/open/pick | sh | fzf, xdg-open/open |
 | [fzplug](fzplug) | Fuzzy find, preview and run other plugins | sh | fzf |
 | [getplugs](getplugs) | Update plugins to installed `nnn` version | sh | curl |
-| [gpg\*](gpg\*) | Encrypt/decrypt files using GPG [✓] | sh | gpg |
+| [gitroot](gitroot) | Cd to the root of current git repo | bash | git |
+| [gpge](gpge) | Encrypt/decrypt files using GPG [✓] | sh | gpg |
```
Comment (PatrickF1) [inline plugins/README.md:33]: this is a typo fix; if you click it currently on master it 404s

File: plugins/gitroot [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+
+# Description: cd to the top level of the current git repository in the current context
+# Shell: bash
```
Comment (N-R-K) [inline plugins/gitroot:4]: What's the reason for requiring `Bash` ? I'm not seeing any bashism in here.

**Review by N-R-K — CHANGES_REQUESTED** [review]

**Comment by PatrickF1** [comment]
Sure I will change it to sh honestly didn’t really know if there were any
bashisms

**Comment by PatrickF1** [comment]
Done! Any other feedback?

**Comment by PatrickF1** [comment]
@N-R-K could I get a re-review please?

File: plugins/gitroot [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env sh
+
+# Description: cd to the top level of the current git repository in the current context
+# Dependencies: git
+# Shell: sh
+# Author: https://github.com/PatrickF1
+
+if git rev-parse --git-dir > /dev/null 2>&1; then
+    root="$(git rev-parse --show-toplevel)"
+    printf "%s" "0c$root" > "$NNN_PIPE"
```
Comment (N-R-K) [inline plugins/gitroot:10]: Could probably avoid calling `git` twice.

```suggestion
root="$(git rev-parse --show-toplevel 2>/dev/null)"
if [ -n "$root" ]; then
    printf "%s" "0c$root" > "$NNN_PIPE"
```

Untested, but I think it should work.

**Comment by N-R-K** [comment]
> @N-R-K could I get a re-review please?

Looks fine to me. Wait for @jarun to take a look, he's usually responsible for merging things.

File: plugins/gitroot (hunk 3 above)
Comment (PatrickF1) [reply plugins/gitroot:10]: Great idea! Let's make it even faster. Let me test it out right now. And thanks for the review :)

Comment (PatrickF1) [reply plugins/gitroot]: tested and it works!

**Comment by jarun** [comment]
Ultra cool! ;)
Thank you for your contribution!

**Comment by jarun** [comment]
Thank you!

**Comment by PatrickF1** [comment]
Thank you for merging my changes @jarun ! And thank you so much for making this awesome tool with a well-thought-out plugin system. I commend your brilliance 💯

**Pull request merged**